tooling/hibernate-gradle-plugin/src/main/java/org/hibernate/orm/tooling/gradle/EnhancementHelper.java

```diff
@@ -150,27 +185,16 @@ private static void writeOutEnhancedClass(byte[] enhancedBytecode, File file, Lo
 			logger.warn( "Problem preparing class file for writing out enhancements [" + file.getName() + "]" );
 		}
 
-		try {
-			FileOutputStream outputStream = new FileOutputStream( file, false );
-			try {
-				outputStream.write( enhancedBytecode );
-				outputStream.flush();
-			}
-			catch (IOException e) {
-				throw new GradleException( "Error writing to enhanced class [" + file.getName() + "] to file [" + file.getAbsolutePath() + "]", e );
-			}
-			finally {
-				try {
-					outputStream.close();
-				}
-				catch (IOException ignore) {
-				}
-			}
+		try ( FileOutputStream outputStream = new FileOutputStream( file, false ) ) {
+			outputStream.write( enhancedBytecode );
+			outputStream.flush();
```

Comment:
Again, the following nio method introduced since JDK7 might be considerable:
```
Files.write( file.toPath(), enhancedBytecode );
```

Reply:
Made this change as well